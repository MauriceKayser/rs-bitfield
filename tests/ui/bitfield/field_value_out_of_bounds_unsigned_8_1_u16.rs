//! Assertion failure: _TYPE_IN_FIELD_0_EXCEEDS_FIELD_SIZE_OF_1_BIT

extern crate alloc;

#[bitfield::bitfield(8)]
struct BitField(#[field(size = 1)] Field); // Can only store values between `0..=1`.

#[derive(Clone, Copy, Debug, bitfield::Field)]
#[repr(u16)]
enum Field {
    F2 = 0x100
}

fn main() {}