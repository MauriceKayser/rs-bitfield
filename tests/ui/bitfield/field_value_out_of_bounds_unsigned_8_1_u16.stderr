error[E0080]: evaluation of constant value failed
 --> tests/ui/bitfield/field_value_out_of_bounds_unsigned_8_1_u16.rs:6:36
  |
6 | struct BitField(#[field(size = 1)] Field); // Can only store values between `0..=1`.
  |                                    ^^^^^ the evaluated program panicked at 'Size of the field type exceeds the specified field size of 1 bit', $DIR/tests/ui/bitfield/field_value_out_of_bounds_unsigned_8_1_u16.rs:6:36
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0271]: type mismatch resolving `<Field as TryFrom<u8>>::Error == u8`
 --> tests/ui/bitfield/field_value_out_of_bounds_unsigned_8_1_u16.rs:6:36
  |
6 | struct BitField(#[field(size = 1)] Field); // Can only store values between `0..=1`.
  |                                    ^^^^^ expected `u8`, found `Infallible`

error[E0277]: the trait bound `Field: TryFrom<u8>` is not satisfied
 --> tests/ui/bitfield/field_value_out_of_bounds_unsigned_8_1_u16.rs:6:32
  |
6 | struct BitField(#[field(size = 1)] Field); // Can only store values between `0..=1`.
  |                                ^^^^-----
  |                                |   |
  |                                |   required by a bound introduced by this call
  |                                the trait `From<u8>` is not implemented for `Field`
  |
  = help: the trait `TryFrom<u8>` is not implemented for `Field`
          but trait `TryFrom<u16>` is implemented for it
  = help: for that trait implementation, expected `u16`, found `u8`
  = note: required for `u8` to implement `Into<Field>`
  = note: required for `Field` to implement `TryFrom<u8>`

error[E0277]: the trait bound `Field: From<u8>` is not satisfied
 --> tests/ui/bitfield/field_value_out_of_bounds_unsigned_8_1_u16.rs:6:36
  |
6 | struct BitField(#[field(size = 1)] Field); // Can only store values between `0..=1`.
  |                                    ^^^^^ the trait `From<u8>` is not implemented for `Field`
  |
  = help: the trait `TryFrom<u8>` is not implemented for `Field`
          but trait `TryFrom<u16>` is implemented for it
  = help: for that trait implementation, expected `u16`, found `u8`
  = note: required for `u8` to implement `Into<Field>`
  = note: required for `Field` to implement `TryFrom<u8>`
